File: 프로그래머스/unrated/181928. 이어 붙인 수/이어 붙인 수.java
class Solution {
    public int solution(int[] num_list) {
        String evenNum = "";
        String oddNum = "";
        for(int item : num_list) {
            if (item % 2 == 0) evenNum += item;
            else oddNum += item;
        }
        return Integer.parseInt(evenNum) + Integer.parseInt(oddNum);
    }
}

File: 프로그래머스/unrated/181928. 이어 붙인 수/이어 붙인 수.js
function solution(num_list) {
    const [oddNumbers, evenNumbers] = num_list.reduce(([odd, even], num) => num % 2 ? [odd.concat(num), even] : [odd, even.concat(num)], [[], []]);
  return parseInt(oddNumbers.join("")) + parseInt(evenNumbers.join(""));
}